android-project/app/src/main/java/oceantreasur/es/fragments/AnimationController.java

```diff
@@ -0,0 +1,135 @@
+import static oceantreasur.es.view.AnimationConstants.MIN_TIME_OFFSET_BIG_FISH;
+import static oceantreasur.es.view.AnimationConstants.MIN_TIME_OFFSET_SMALL_FISH;
+import static oceantreasur.es.view.AnimationConstants.SPAWNING_POINT_OF_BIG_FISH;
+import static oceantreasur.es.view.ScreenUtils.getDimensionOfFishInPx;
+import static oceantreasur.es.view.ScreenUtils.getScreenHeight;
+import static oceantreasur.es.view.ScreenUtils.getScreenWidth;
+
+
+public class AnimationController {
+
+    private ArrayList<View> smallFishViews = new ArrayList<>();
+    private ArrayList<View> bigFishViews = new ArrayList<>();
+    private int smallFishBeginOffset = getDimensionOfFishInPx(small_fish_width);
+    private int bigFishBeginOffset = getDimensionOfFishInPx(R.dimen.big_fish_width);
+
+    AnimationController(RelativeLayout background) {
+        inflateFishViews(background);
+        moveAnimation();
+    }
+
+    private void inflateFishViews (RelativeLayout background) {
+
+        LayoutInflater Li = LayoutInflater.from(OceanTreasuresApplication.getStaticContext());
+
+        for(int i = 0; i < MAX_SMALL_FISH; i++) {
+            smallFishViews.add(Li.inflate(small_fish, null));
+            background.addView(smallFishViews.get(i));
+        }
+
+        for(int i = 0; i < MAX_BIG_FISHES; i++) {
+            bigFishViews.add(Li.inflate(big_fish, null));
+            background.addView(bigFishViews.get(i));
+        }
+    }
+
+    private void moveAnimation() {
+        setupAnimation(smallFishViews, smallFishBeginOffset);
+        setupAnimation(bigFishViews, bigFishBeginOffset);
+    }
+
+    private void setupAnimation(ArrayList<View> views, int offset) {
+        for(int i = 0; i < views.size(); i++) {
+            startAnimation(views.get(i), generateRandomIntegerInRange(MIN_DURATION, MAX_DURATION), offset);
+        }
+    }
+
+    private int choseRandomYPosition() {
+        int height = getScreenHeight();
+
+        int yPosition = generateRandomIntegerInRange(1, height);
+
+        if(yPosition > height * MAX_SPAWNING_POINT_IN_PERCENTS) {
+            yPosition -= (getDimensionOfFishInPx(small_fish_height) + FISH_MARGIN_BOTTOM);
+        } else {
+            if(yPosition < height * MIN_SPAWNING_POINT_IN_PERCENTS) {
+                yPosition += (getDimensionOfFishInPx(small_fish_height) - FISH_MARGIN_TOP);
+            }
+        }
+
+        return yPosition;
+    }
+
+    private void startAnimation(final View fish, final int durationTime, final int beginOffset) {
+        int width = getScreenWidth();
+        int height;
+        boolean isSmallFish = true;
+
+        if(smallFishBeginOffset ==  beginOffset) {
+            height = choseRandomYPosition();
+        } else {
+            isSmallFish = false;
+            height = SPAWNING_POINT_OF_BIG_FISH;
+        }
+
+        Animation anim = new TranslateAnimation(-beginOffset, width + beginOffset, height, height);
+        anim.setFillAfter(true);
+
+        if(isSmallFish) {
```

Comment:
Remove if statement. The logic should be the same for any fish. Pass time offsets and duration as parameters of the method. Bundle these three variables into a small model (AnimationTiming for example). Have two concrete objects, used for all fish - smallFishAnimationTiming and bigFishAnimationTiming.